runtime/vm/jvminit.c

```diff
@@ -6000,6 +6000,270 @@ processCompressionOptions(J9JavaVM *vm){
+	while (xlogindex >= 0) {
+		BOOLEAN unrecognizedOption = FALSE;
+		BOOLEAN setgclog = FALSE;
+		BOOLEAN gclog = FALSE;
+		char *gclogfile = NULL;
+		char xlogoptionsbuf[LARGE_STRING_BUF_SIZE];
+		char *xlogoptions = xlogoptionsbuf;
+		/* Include room for the trailing NULL */
+		UDATA optionLen = strlen(j9vm_args->actualVMArgs->options[xlogindex].optionString) + 1;
+
+		if (optionLen > sizeof(xlogoptionsbuf)) {
+			xlogoptions = j9mem_allocate_memory(optionLen, OMRMEM_CATEGORY_VM);
+			if (NULL == xlogoptions) {
+				goto xlogret;
+			}
+		}
+
+		if (OPTION_OK != COPY_OPTION_VALUE(xlogindex, ':', &xlogoptions, optionLen)) {
+			rc = JNI_ERR;
+			goto xlogret;
+		}
+
+		if (NULL == xlogoptions) {
+			/* Reached if just -Xlog is specified */
+			setgclog = TRUE;
+			gclog = TRUE;
+			unrecognizedOption = TRUE;
+		} else {
+			/* -Xlog:gc+other=off,a+b=c:file=abc:dec:options */
+			char *fullOptions = xlogoptions;
+			char *upToColon = strsep(&fullOptions, ":");
+#if defined(DEBUG_XLOG)
+			j9tty_printf(PORTLIB, "upToColon %s fullOptions %s\n", upToColon, fullOptions);
+#endif /* defined(DEBUG_XLOG) */
+			if ('\0' == *upToColon) {
+				/* Nothing specified means all are enabled */
+				setgclog = TRUE;
+				gclog = TRUE;
+				unrecognizedOption = TRUE;
+			} else {
+				char *upToComma = strsep(&upToColon, ",");
+#if defined(DEBUG_XLOG)
+				j9tty_printf(PORTLIB, "upToComma %s upToColon %s\n", upToComma, upToColon);
+#endif /* defined(DEBUG_XLOG) */
+				while (TRUE) {
+					if ('\0' == *upToComma) {
+						/* Missing tag */
+xlogerr:
+						CONSUME_ARG(j9vm_args, xlogindex);
+						j9nls_printf(PORTLIB, J9NLS_ERROR, J9NLS_VM_OPTION_MALFORMED, j9vm_args->actualVMArgs->options[xlogindex].optionString);
+						rc = JNI_ERR;
+						goto xlogret;
+					}
+					BOOLEAN checkTagLevel = FALSE;
+					char *upToEquals = strsep(&upToComma, "=");
+#if defined(DEBUG_XLOG)
+					j9tty_printf(PORTLIB, "upToEquals %s upToComma %s\n", upToEquals, upToComma);
+#endif /* defined(DEBUG_XLOG) */
+					if ('\0' == *upToEquals) {
+						/* Missing tag */
+						goto xlogerr;
+					} else {
+						UDATA numtags = 0;
+						char *tag = strsep(&upToEquals, "+");
+#if defined(DEBUG_XLOG)
+						j9tty_printf(PORTLIB, "tag %s upToEquals %s\n", tag, upToEquals);
+#endif /* defined(DEBUG_XLOG) */
+						while (TRUE) {
+							if ('\0' == *tag) {
+								/* Missing tag */
+								goto xlogerr;
+							}
+							if (0 == j9_cmdla_stricmp("all", tag)) {
+								if ((numtags > 0) || (NULL != upToEquals)) {
+									/* Invalid use of keyword all */
+									goto xlogerr;
+								} else {
+									checkTagLevel = TRUE;
+									/* only gc is recognized from all */
+									unrecognizedOption = TRUE;
+								}
+							} else if (0 == j9_cmdla_stricmp("disable", tag)) {
+								if ((numtags > 0) ||
+									(NULL != upToEquals) ||
+									(NULL != upToComma) ||
+									(NULL != upToColon) ||
+									(NULL != fullOptions)
+								) {
```

Comment:
Please move each `||` to the beginning of the next line; see https://github.com/eclipse/omr/blob/master/doc/CodingStandard.md.